### PR TITLE
Created list contacts UI

### LINKED ISSUE #6: Create list contacts UI
- [x] Create a contacts page
- [x] Add a table with dummy list of contacts
- [x] Add a search bar and Add Contact button at the top.

### LINKED ISSUE #13: Update the review changes for note creation feature
- [ ] Replace Ramda with filter function https://github.com/jenipharachel/neeto-ui-challenge/pull/12#discussion_r1098532783

### PR DESCRIPTION
**Pull request by jenipharachel**
Fixes #6, Fixes #13 

@yedhink _a Please review

- Added a screen to list the contacts in table format
- Added functionality to delete contacts
- Replaced the Ramda function with the filter function in the logic for deleting notes 
- Turned the DeleteAlert component and MenuBar component into reusable components

Screen recording:

https://user-images.githubusercontent.com/35962711/217311957-bf6b6f81-06f1-4c16-8ef4-e5e059f88957.mov

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Contacts/Table.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      render: renderText,
+    },
+    {
+      title: "",
+      dataIndex: "icon_button",
+      key: "icon_button",
+      width: "10%",
+      render: (_, { id }) => renderDropdown({ onDeleteContact, id }),
+    },
+  ];
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Table.jsx:39]: Why's this constant kept within the presentation layer? Move this to a `constants` file within the closest scope.

Refer: https://academy.bigbinary.com/learn-react/react-best-practices/structuring-code

File: app/javascript/src/components/Dashboard/Contacts/index.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    setSelectedContactId(id);
+    setShowDeleteAlert(true);
+  };
+
+  return (
+    <div className="max-w-fit flex">
+      <MenuBar menuBarOptions={MENU_BAR_OPTIONS} showMenu={showMenu} />
+      <Container>
+        <Header
+          menuBarToggle={() => setShowMenu(!showMenu)}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/index.jsx:41]: https://academy.bigbinary.com/learn-react/react-best-practices/know-when-to-use-previous-state-for-state-update

File: app/javascript/src/components/Dashboard/Contacts/utils.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+
+import { MenuHorizontal } from "@bigbinary/neeto-icons";
+import { Avatar, Dropdown, Typography } from "@bigbinary/neetoui";
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/utils.jsx:4]: Create aliases for commonly used paths. https://academy.bigbinary.com/learn-react/react-best-practices/create-alias-for-commonly-used-directories

**Review by yedhink — COMMENTED** [review]
@jenipharachel

File: app/javascript/src/components/Dashboard/Contacts/Table.jsx (hunk 1 above)
Comment (jenipharachel) [reply app/javascript/src/components/Dashboard/Contacts/Table.jsx:39]: I initially did keep it in the constants file but the UI broke as the id of contact is being passed in line 37 `render: (_, { id }) => renderDropdown({ onDeleteContact, id }),`. I'll see if something else can be done.

File: app/javascript/src/components/Dashboard/Contacts/utils.jsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+const { Menu, MenuItem } = Dropdown;
+
+export const renderContactDetails = name_and_role => {
+  const { name, role } = name_and_role;
```
[the remark this answers is not in the document]
Comment (jenipharachel) [reply app/javascript/src/components/Dashboard/Contacts/utils.jsx:9]: I used the same variable that I used for the keys used in the CONTACTS_TABLE_ROW_DATA. I'll update it with camelCase

File: app/javascript/src/components/Dashboard/Contacts/utils.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  return (
+    <div className="flex flex-row items-center">
+      <Avatar className="mr-2" size="large" user={{ name }} />
+      <div>
+        <Typography style="h5">{name}</Typography>
+        <Typography style="body3">{role}</Typography>
+      </div>
```
[the remark this answers is not in the document]
Comment (jenipharachel) [reply app/javascript/src/components/Dashboard/Contacts/utils.jsx:17]: Yes, it's just a wrapper, will replace it with `span`

File: app/javascript/src/components/Dashboard/Contacts/utils.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (yedhink) [reply app/javascript/src/components/Dashboard/Contacts/utils.jsx:9]: >I used the same variable that I used for the keys used in the CONTACTS_TABLE_ROW_DATA

@jenipharachel Please understand that, it's not the right way to deal with things. Naming conventions in BigBinary are documented for a reason. Ensure that you stick onto them or raise concerns if you feel that the convention doesn't make sense.

File: app/javascript/src/components/Dashboard/Contacts/Table.jsx (hunk 1 above)
Comment (jenipharachel) [reply app/javascript/src/components/Dashboard/Contacts/Table.jsx:39]: @yedhink https://youtu.be/4pnhKd41Cn0

Comment (jenipharachel) [reply app/javascript/src/components/Dashboard/Contacts/Table.jsx:39]: @yedhink I fixed it by storing the return value of the function `COLUMN_DATA` from constants file to a variable. Then, providing the variable to the columnData param in NeetoUITable.

**Pull request merged**